Fixed theme + migrated to newer Hugo features

kdevo opened the pull request:
Hi there,

first: Thanks for the cool theme!

I just applied the needed changes for #69 and got this theme working again.

Additionally, I did the following improvements :sparkles::
 
- Implement native cache busting via the newer [Hugo Pipes](https://gohugo.io/hugo-pipes):
  - Use fingerprinting for the resource filenames
  - No additional setup needed anymore
  - Move to the newer `assets` directory:
    - Use the SCSS (sass) sources and convert them to CSS using `resources.ToCSS`
    - Use `resources.Minify`
    - Remove minified files where possible and only keep the sources
- Remove Hugo linter warnings:
  - Use `.Site.RegularPages` instead of `.Data.Pages` where needed
  - Use `.File.UniqueID` instead of `.UniqueID`
  - Use `{{ with .OutputFormats.Get "RSS" }} ...`
  - Remove deprecated Hugo generator meta tag (will be injected automatically, anyways)
- Pin version to Hugo `0.5` (maybe `0.43` would do the job, too)

## **:warning: Please test this PR thoroughly before merging into master.**
I just **rapidly** wanted to share my changes and help out a bit, but didn't have the time to test everything.

kdevo committed:
:zap: :pushpin: :bug: Integrate newer Hugo features, pin theme to Hugo v0.5

kdevo committed:
:bug: Re-add accidently (re)moved content files

kdevo committed:
:pushpin: :bug: Fix min_version (needs to be a string)

kdevo committed:
:bug: Pipes: Change order - first minify, then fingerprint

tomanistor commented:
I appreciate the PR @kdevo! Let me test it out and I'll merge it in if it works.

kdevo committed:
:bug: Fix contact form; ExecuteAsTemplate

This bug came up while developing git.io/osprey-delight.
So I decided to fix it here, too.

kdevo commented:
Just a note: As [already commented](https://github.com/tomanistor/osprey/issues/69#issuecomment-557877734) in the yet closed #69 I [forked](https://git.io/osprey-delight) this theme but still wanted to fix the above bug in the original merge request regarding the contact form functionality.
This was the only bug I encountered.

tomanistor commented:
Thanks for the updates @kdevo! I've just tested on my own site and merged into master.
